src/formatters/source/pushResultFormatter.ts

```diff
@@ -85,7 +85,14 @@ export class PushResultFormatter extends ResultFormatter {
 
     // Throw a DeployFailed error unless the deployment was successful.
     if (!this.isSuccess()) {
-      throw new SfError(messages.getMessage('sourcepushFailed'), 'PushFailed');
+      // Add error message directly on the DeployResult (e.g., a GACK)
+      let errMsg = '';
+      this.results?.forEach((res) => {
+        if (res.response?.errorMessage) {
+          errMsg += `${res.response?.errorMessage}\n`;
+        }
+      });
```

Comment:
Is `\n` fine across os ?  Or do you need the EOL thing? 

```suggestion
      const errMsg = this.results?.filter((res) => res.response?.errorMessage).join('\n')
```